### PR TITLE
Fix escaping of dollar symbols in HTMLWriter

### PR DESCRIPTION
**Pull request by odow**
Previously, Javascript LaTeX engines got confused trying to
distinguish between inline delimiters and dollar symbols for actual
currency. This works around the problem by wrapping (currency) dollar
symbols in a span tag.

Closes https://github.com/JuliaDocs/Documenter.jl/issues/890

I could use come pointers on how to test this properly. (Is there a way to test the output of the rendered HTML?) I got confused with the structure of `/test`. Here's my more-involved local test:

```julia
shell> cat src/index.md
The price is \$1.

The price is \$1 or \$2.

Here is some $x^2$ math.

The pr\$ice is \$s for some $x^2$.

julia> Documenter.makedocs(sitename = "Bug", pages = ["index.md"])
[ Info: SetupBuildDirectory: setting up build directory.
[ Info: Doctest: running doctests.
[ Info: ExpandTemplates: expanding markdown templates.
[ Info: CrossReferences: building cross-references.
[ Info: CheckDocument: running document checks.
[ Info: Populate: populating indices.
[ Info: RenderDocument: rendering document.
[ Info: HTMLWriter: rendering HTML pages.
```

image

### CODE REVIEW

File: test/htmlwriter.jl [hunk 1]
```diff
@@ -171,5 +171,10 @@ end
             end
         end
     end
+
+    @test "Dollar escapes" begin
```
Comment (mortenpi) [inline test/htmlwriter.jl:175]:
```suggestion
    @testset "Dollar escapes" begin
```

Comment (odow) [reply test/htmlwriter.jl]: Oops. My mistake

**Review by mortenpi — APPROVED** [review]
LGTM, I think this is a clever workaround. And doesn't have the potentially breaking aspect of changing the delimiter configuration.

We don't really have good integration tests of the outputs. The best we do currently are the `occursin`s [in the example builds](https://github.com/JuliaDocs/Documenter.jl/blob/master/test/examples/tests.jl). Potentially you could add a few tests there -- I think having your snippet in the `test/examples` build would be good.

Just a passing thought, beyond the scope of this PR, but in the future we could use [Gumbo](https://github.com/JuliaWeb/Gumbo.jl) to parse the generated HTML and then do some sanity checks on the AST. It worked well for generating the outdated version warnings.

**Comment by odow** [comment]
@mortenpi how's this?

**Comment by mortenpi** [comment]
Looks good @odow, thanks!

**Pull request merged**